[C#] Fix source generation of statically imported members

PavielKraskouski opened the pull request:
Fixes #93666
Possibly also fixes #85844

Adds `using static` declarations to the beginning of generated C# scripts for default values. Enables the use of static imports when initializing properties and fields.

PavielKraskouski committed:
Fix source generation of statically imported members

paulloz commented:
Hello, and thank you for your contribution!

I don't think this is the right way to go about fixing this issue. We want to avoid injecting `using static` statements into our generated sources, and would rather have the values fully qualified properly. I thought it should already be the case (both for fields[^1] and properties[^2]), so I had a quick glance. It looks like `SymbolDisplayFormat.FullyQualifiedFormat`[^3] does not qualify this how I'd expect because it is missing `SymbolDisplayMemberOptions.IncludeContainingType`.

I think this is a more proper route to fix the linked issues than adding `using static` statements in the generated sources. It would also be nice to have a new test case covering this, to avoid potential future regressions.

[^1]: [ScriptPropertyDefValGenerator.cs#L334-L340](https://github.com/godotengine/godot/blob/f192430/modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/ScriptPropertyDefValGenerator.cs#L334-L340)
[^2]: [ScriptPropertyDefValGenerator.cs#L211-L276](https://github.com/godotengine/godot/blob/f192430/modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/ScriptPropertyDefValGenerator.cs#L211-L276)
[^3]: We use it [here](https://github.com/godotengine/godot/blob/f192430/modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/ExtensionMethods.cs#L184-L190)

PavielKraskouski commented:
@paulloz Hello! Added `SymbolDisplayMemberOptions.IncludeContainingType` instead of `using static` directives. I also discovered and fixed a bug where incorrect code was generated if a static import was used inside an expression. For example, `RadToDeg(2 * Pi)` was generated as `global::Godot.Mathf.RadToDeg(2 * Pi)`. Consequently, the compiler didn't know what `Pi` was.

paulloz reviewed (commented):
Looks good to me, thanks for the changes! You should probably re-word the commit message to reflect what this does, and I'll be able to approve 👍
